### PR TITLE
Added pagination in API

### LINKED ISSUE #385: Add paginated responses to api v2
API v1 used to have paginated views which would give 20 items per page. 
Is the same needed to be done in API v2 ?
cc @juslee @shivamMg @leto 

EDIT - See https://github.com/fossasia/open-event-orga-server/issues/385#issuecomment-221485640

### PR DESCRIPTION
**Pull request by aviaryan**
Fixes #385 
See https://github.com/fossasia/open-event-orga-server/issues/385#issuecomment-221485640 for more details.

### CODE REVIEW

File: open_event/api/utils.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    'start': fields.Integer,
+    'limit': fields.Integer,
+    'count': fields.Integer,
+    'next': fields.String,
+    'previous': fields.String
+})
+
+
+# Base class for Paginated Resource
+class PaginatedResourceBase():
```
Comment (juslee) [inline open_event/api/utils.py]: Codacy Issue found: [Too few public methods (0/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554311/issues/source?bid=3233864&fileBranchId=3318429#l15)

File: open_event/api/sessions.py [hunk 2]
```diff
@@ -70,3 +76,19 @@ def get(self, event_id):
         get_object_or_404(EventModel, event_id)
 
         return get_object_list(SessionModel, event_id=event_id)
+
+
+@api.route('/events/<int:event_id>/sessions/page')
```
Comment (juslee) [inline open_event/api/sessions.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554307/issues/source?bid=3233864&fileBranchId=3318429#l81)

File: open_event/api/formats.py [hunk 3]
```diff
@@ -35,3 +42,19 @@ def get(self, event_id):
         get_object_or_404(EventModel, event_id)
 
         return get_object_list(FormatModel, event_id=event_id)
+
+
+@api.route('/events/<int:event_id>/formats/page')
```
Comment (juslee) [inline open_event/api/formats.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554302/issues/source?bid=3233864&fileBranchId=3318429#l47)

File: open_event/api/speakers.py [hunk 4]
```diff
@@ -51,3 +57,19 @@ def get(self, event_id):
         get_object_or_404(EventModel, event_id)
 
         return get_object_list(SpeakerModel, event_id=event_id)
+
+
+@api.route('/events/<int:event_id>/speakers/page')
```
Comment (juslee) [inline open_event/api/speakers.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554308/issues/source?bid=3233864&fileBranchId=3318429#l62)

File: open_event/api/events.py [hunk 5]
```diff
@@ -39,3 +44,16 @@ class EventList(Resource):
     def get(self):
         """List all events"""
         return get_object_list(EventModel)
+
+
+@api.route('/page')
```
Comment (juslee) [inline open_event/api/events.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554301/issues/source?bid=3233864&fileBranchId=3318429#l49)

File: open_event/api/languages.py [hunk 6]
```diff
@@ -36,3 +42,19 @@ def get(self, event_id):
         get_object_or_404(EventModel, event_id)
 
         return get_object_list(LanguageModel, event_id=event_id)
+
+
+@api.route('/events/<int:event_id>/languages/page')
```
Comment (juslee) [inline open_event/api/languages.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554304/issues/source?bid=3233864&fileBranchId=3318429#l47)

File: open_event/api/sponsors.py [hunk 7]
```diff
@@ -36,3 +42,19 @@ def get(self, event_id):
         get_object_or_404(EventModel, event_id)
 
         return get_object_list(SponsorModel, event_id=event_id)
+
+
+@api.route('/events/<int:event_id>/sponsors/page')
```
Comment (juslee) [inline open_event/api/sponsors.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554309/issues/source?bid=3233864&fileBranchId=3318429#l47)

File: open_event/api/levels.py [hunk 8]
```diff
@@ -35,3 +41,19 @@ def get(self, event_id):
         get_object_or_404(EventModel, event_id)
 
         return get_object_list(LevelModel, event_id=event_id)
+
+
+@api.route('/events/<int:event_id>/levels/page')
```
Comment (juslee) [inline open_event/api/levels.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554305/issues/source?bid=3233864&fileBranchId=3318429#l46)

File: open_event/api/microlocations.py [hunk 9]
```diff
@@ -39,3 +45,19 @@ def get(self, event_id):
         get_object_or_404(EventModel, event_id)
 
         return get_object_list(MicrolocationModel, event_id=event_id)
+
+
+@api.route('/events/<int:event_id>/microlocations/page')
```
Comment (juslee) [inline open_event/api/microlocations.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554306/issues/source?bid=3233864&fileBranchId=3318429#l50)

File: open_event/api/tracks.py [hunk 10]
```diff
@@ -41,3 +47,19 @@ def get(self, event_id):
         get_object_or_404(EventModel, event_id)
 
         return get_object_list(TrackModel, event_id=event_id)
+
+
+@api.route('/events/<int:event_id>/tracks/page')
```
Comment (juslee) [inline open_event/api/tracks.py]: Codacy Issue found: [Too few public methods (1/2)](https://www.codacy.com/app/triplez/open-event-orga-server/file/2632554310/issues/source?bid=3233864&fileBranchId=3318429#l52)

**Comment by aviaryan** [comment]
@juslee Please review. Will write tests for it tomorrow.

**Pull request merged**